pennylane/_qubit_device.py

```diff
@@ -173,9 +178,17 @@ def _permute_wires(self, observable):
         "Projector",
     }
 
-    def __init__(self, wires=1, shots=None, analytic=None):
+    def __init__(
+        self, wires=1, shots=None, *, r_dtype=np.float64, c_dtype=np.complex128, analytic=None
+    ):
         super().__init__(wires=wires, shots=shots, analytic=analytic)
 
+        assert "float" in str(r_dtype)
+        assert "complex" in str(c_dtype)
```

Comment:
Instead of asserts, maybe it would be good to have some errors with descriptive error messages here such that it's apparent to users what went wrong.

Reply:
Changed to raise a `ValueError`, but do we have better exceptions for this case?

Reply:
Oh, we have `qml.DeviceError`, that could work maybe? Not more specific than that. I think we'd aiming at introduce more specific errors in PennyLane, though it may be out of scope here (unless we'd like a DtypeMismatchError or so).

Reply:
Okay. It now raises `qml.DeviceError`.